src/Underworld.ts

```diff
@@ -3558,6 +3558,10 @@ ${CardUI.cardListToImages(player.stats.longestSpell)}
     const closest = sortedByProximityToCoords[0]
     return closest;
   }
+  getUnitById(id: number, prediction: boolean): Unit.IUnit | undefined {
+    const units = prediction ? this.unitsPrediction : this.units;
+    return units.find(u => u.id == id);
+  }
```

Comment:
I foresee potential issues with this function given that prediction units have inconsistent ids.  Has this been tested with clone / split prediction units?

Reply:
Leaving this here per our discord convo:

> I'd just say test it thoroughly (like clone and then bloat and make sure it passes the correct sourceUnit) and if it works fine, then we can merge it
(...)
I see a potential problem as a prediction unit getting an id, the id gets saved in the modifier, and then the prediction unit gets cleaned up or something
(...)
"use .real instead of assuming that prediction units id matched their real counterparts id"
Maybe just leave a comment to that effect with getUnitById